chalk-solve/src/wf.rs

```diff
@@ -486,29 +492,233 @@ fn compute_assoc_ty_goal<I: Interner>(
     ))
 }
 
-/// Computes a goal to prove Sized constraints on a struct definition.
-/// Struct is considered well-formed (in terms of Sized) when it either
-/// has no fields or all of it's fields except the last are proven to be Sized.  
-fn compute_struct_sized_constraint<I: Interner>(
-    db: &dyn RustIrDatabase<I>,
-    fields: &[Ty<I>],
-) -> Option<Goal<I>> {
-    if fields.len() <= 1 {
-        return None;
+struct WellKnownConstraints {}
+
+impl WellKnownConstraints {
+    pub fn inside_impl<I: Interner>(
+        db: &dyn RustIrDatabase<I>,
+        trait_ref: &TraitRef<I>,
+    ) -> Option<Goal<I>> {
+        match db.trait_datum(trait_ref.trait_id).well_known? {
+            WellKnownTrait::CopyTrait => Self::copy_impl_constraint(db, trait_ref),
+            WellKnownTrait::DropTrait | WellKnownTrait::CloneTrait | WellKnownTrait::SizedTrait => {
+                None
+            }
+        }
     }
 
-    let interner = db.interner();
+    pub fn outside_impl<I: Interner>(
+        db: &dyn RustIrDatabase<I>,
+        impl_datum: &ImplDatum<I>,
+    ) -> Option<Goal<I>> {
+        let interner = db.interner();
+
+        match db.trait_datum(impl_datum.trait_id()).well_known? {
+            // You can't add a manual implementation of Sized
+            WellKnownTrait::SizedTrait => Some(GoalData::CannotProve(()).intern(interner)),
```

Comment:
This seems like it will ultimately have to change, since the resulting UX would be terrible, no?

Reply:
By UX you mean writing tests and using repl?

Reply:
`CannotProve` is not the best way to report errors, but it I guess we'll change that when we add proper error reporting

Reply:
By UX I mean "when rustc runs this code" -- the vision eventually is that rustc will run these WF goals and report errors from the results. But I agree we can address it later.